xapian-core/backends/glass/glass_changes.cc

```diff
@@ -64,7 +65,9 @@ GlassChanges::start(glass_revision_number_t old_rev,
     // Always check max_changesets for modification since last revision.
     const char *p = getenv("XAPIAN_MAX_CHANGESETS");
     if (p) {
-	max_changesets = atoi(p);
+	if (!parse_signed(p, max_changesets)) {
+	    throw "XAPIAN_MAX_CHANGESETS must be a positive integer";
+	}
```

Comment:
Currently setting `XAPIAN_MAX_CHANGESETS` to an empty value has the same effect as not setting it, which seems desirable behaviour to keep - an environment variable being empty works the same as it being unset in many contexts, so it's something people might reasonably be relying on.  Making the condition `(p && *p)` should work.

Reply:
I think this should use `parse_unsigned` so negative values are actually rejected (they don't make sense, and the error message implies they aren't allowed).  `0` is OK though (it's the default after all) so the message should say `non-negative` for clarity.

Reply:
fixed in a fixup commit.